src/Menu/NotificationManager.php

```diff
@@ -0,0 +1,196 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Menu;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Registerable;
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\Admin\PageController;
+
+/**
+ * Class NotificationManager
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Menu
+ *
+ * Manages the display of a single, aggregated notification pill in the admin menu.
+ * It relies on a filter to gather the total count from various contributors.
+ */
+class NotificationManager implements Service, Registerable {
+
+	/**
+	 * Register the service, hooking into admin_menu to display notifications.
+	 */
+	public function register(): void {
+		// Hook into admin_menu with a high priority (e.g., 20) to ensure
+		// all other menu items have been registered by WooCommerce and other plugins.
+		add_action( 'admin_menu', [ $this, 'display_aggregated_notification_pill' ], 20 );
+
+		// Persist the pill across menu switches between the Marketing and Analytics menus.
+		add_action( 'admin_footer', [ $this, 'persist_pill' ], 10 );
+	}
+
+	/**
+	 * Determines if the current admin page is a child page within the WooCommerce Marketing section.
+	 * This logic is crucial for deciding where the notification pill should be placed.
+	 *
+	 * @return bool True if the current page is a Marketing child page, false otherwise.
+	 */
+	private function is_marketing_page() {
+		global $pagenow;
+
+		$current_page_slug = isset( $_GET['page'] ) ? sanitize_text_field( wp_unslash( $_GET['page'] ) ) : ''; // phpcs:ignore WordPress.Security.NonceVerification
+		$current_page_path = isset( $_GET['path'] ) ? sanitize_text_field( wp_unslash( $_GET['path'] ) ) : ''; // phpcs:ignore WordPress.Security.NonceVerification
+		$current_post_type = isset( $_GET['post_type'] ) ? sanitize_text_field( wp_unslash( $_GET['post_type'] ) ) : ''; // phpcs:ignore WordPress.Security.NonceVerification
+
+		$post_type_page = add_query_arg(
+			[
+				'post_type' => $current_post_type,
+			],
+			$pagenow
+		);
+
+		$page_path_fragment = str_replace(
+			'page=',
+			'',
+			build_query(
+				[
+					'page' => $current_page_slug,
+					'path' => $current_page_path,
+				]
+			)
+		);
+
+		$page_controller_pages = PageController::get_instance()->get_pages();
+		$marketing_menu_slug   = 'woocommerce-marketing';
+
+		$marketing_menu_pages = array_filter(
+			$page_controller_pages,
+			function ( $page ) use ( $marketing_menu_slug ) {
```

Comment:
```suggestion
			static function ( $page ) use ( $marketing_menu_slug ) {
```

Use `static` that improve performance.